BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -966,6 +973,13 @@ CalculateResource (
   UINT64                      Base;
   UINT64                      Alignment;
 
+  if ((BarType == PciBarTypeUnknown) || \
+      (BarType == PciBarTypeIo) || \
+      (BarType == PciBarTypeMem) || \
+      (BarType >= PciBarTypeMaxType)) {
+    return;
+  }

```

Comment:
Why do we do this ?

Reply:
Klocwork reports out of bound index since sizeof (PCI_BAR_TYPE) 9 is greater than PCI_MAX_BAR 6.
Array 'PciBar' of size 6 may use index value(s) -1
Array 'PciBar' of size 6 may use index value(s) 6..8

Reply:
Got it.
should we jus t do   if ((BarType == PciBarTypeUnknown) ||  ((BarType >= PciBarTypePMem64) ?

Reply:
Let me update the condition as recommened. Thanks.